<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1222</width>
    <height>750</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Metamorphic Testing Tool for Image Recognition System</string>
  </property>
  <property name="styleSheet">
   <string notr="true">*{	
	font-family:微软雅黑;
	font-size:15px;
	color: white;
}
QWidget {
background-color:#17233d;
}
#path_lineEdit{
	width:80px;
	height:30px;
	font-size:15px;
	font-family:微软雅黑;
	color:rgb(73, 232, 234);
}


#select_folder_button{
	width:80px;
	height:40px;
	font-family:微软雅黑;
	font-size:12px;

}



Line{
	border:1px dashed #1d649c;
}

QTextEdit{
   border:1px solid #1d649c;
}
QCheckBox{
    spacing: 2px;
    color: white;
}
QPushButton{
	border:-1px solid #1e92bc;
	background-color: rgba(0, 153, 255, 0.3);

}


#input_path_label{
	width:80px;
	height:30px;
	color:#6fdcff;
	font-family:微软雅黑;
	font-size:15px;
}
#fusion_mutation_button,#seg_mutation_button,#gen_deu_img_button,#test_exection_button{
	color:#6fdcff;
	font-size:15px;

}
#fusion_mutation_button,#seg_mutation_button {	
	width:80px;
	height:40px;
	font-family:微软雅黑;
	font-size:14px;
	color:white;
}


#test_exection_button,#gen_deu_img_button,#gen_result_button {	
	width:140px;
	height:40px;
	font-family:微软雅黑;
	font-size:14px;
	color:white;
}

</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_4" stretch="1,4,6">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_8" stretch="7,3">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <item>
        <widget class="QLabel" name="input_path_label">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="font">
          <font>
           <family>微软雅黑</family>
           <pointsize>-1</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Input Path of Original Test Data：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="path_lineEdit">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="select_folder_button">
         <property name="text">
          <string>Select Folder</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGroupBox" name="data_mutation_module">
       <property name="title">
        <string>Test Data Augmentation Module</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="seg_mutation_button">
          <property name="font">
           <font>
            <family>微软雅黑</family>
            <pointsize>-1</pointsize>
            <weight>50</weight>
            <bold>false</bold>
           </font>
          </property>
          <property name="text">
           <string>Segmentation
 Mutation</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="fusion_mutation_button">
          <property name="font">
           <font>
            <family>微软雅黑</family>
            <pointsize>-1</pointsize>
            <weight>50</weight>
            <bold>false</bold>
           </font>
          </property>
          <property name="text">
           <string>Rainy Effect</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="test_execution_module">
     <property name="title">
      <string>Test Execution Module</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,1,0,2,0,0,1,1,1">
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1,1,0,1,1">
        <item>
         <widget class="QCheckBox" name="MT1_checkbox">
          <property name="text">
           <string>MR1：Brightness   parameter:[-100,100],step:20</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">checkbox_group</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="MT2_checkbox">
          <property name="text">
           <string>MR2：Rotation   parameter:[10,100],step:10</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">checkbox_group</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="MT3_checkbox">
          <property name="text">
           <string>MR3：Translation   parameter:from(20,20)to(110,110),step:10</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">checkbox_group</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="MT4_checkbox">
          <property name="text">
           <string>MR4：Cropping   parameter:[1,10],step:1</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">checkbox_group</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="MT5_checkbox">
          <property name="text">
           <string>MR5：Scale   parameter:[0.15,1.75],step:0.15</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">checkbox_group</string>
          </attribute>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_6">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,2">
        <item>
         <widget class="QCheckBox" name="MT6_checkbox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>MR6：Blur     </string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">checkbox_group</string>
          </attribute>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QLabel" name="gaussian_label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Gaussian   parameter:3*3,5*5,7*7</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="averaging_label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Averaging   parameter:3*3,4*4,5*5,6*6</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="Median_label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Median    parameter:3,5</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="bilateral_filter_label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Bilateral Filter    parameter:(9,75,75)</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QCheckBox" name="all_MT_checkbox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <family>微软雅黑</family>
            <pointsize>-1</pointsize>
            <weight>75</weight>
            <italic>false</italic>
            <bold>true</bold>
           </font>
          </property>
          <property name="layoutDirection">
           <enum>Qt::RightToLeft</enum>
          </property>
          <property name="text">
           <string>Select all MRs     </string>
          </property>
          <property name="tristate">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="gen_deu_img_button">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <family>微软雅黑</family>
            <pointsize>-1</pointsize>
            <weight>50</weight>
            <bold>false</bold>
           </font>
          </property>
          <property name="text">
           <string>Generate Follow-up
Test Data</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="test_exection_button">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <family>微软雅黑</family>
            <pointsize>-1</pointsize>
            <weight>50</weight>
            <bold>false</bold>
           </font>
          </property>
          <property name="text">
           <string>Execute Testing</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="6,4">
     <item>
      <widget class="QGroupBox" name="test_judgement_module">
       <property name="title">
        <string>Test Oracle Checking Module</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Original and Follow-up Image Test Results:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextBrowser" name="ori_deu_res_textBrowser">
            <property name="placeholderText">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Metamorphic Testing Results:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextBrowser" name="mt_res_textBrowser">
            <property name="placeholderText">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="test_result_module">
       <property name="title">
        <string>Test Result Analysis Module</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QSplitter" name="splitter">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="QPushButton" name="gen_MR_res_button">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>458</width>
             <height>24</height>
            </size>
           </property>
           <property name="text">
            <string>Generate Metamorphic Testing Analysis Results</string>
           </property>
          </widget>
          <widget class="QTextBrowser" name="MT_result_textBrowser">
           <property name="placeholderText">
            <string/>
           </property>
          </widget>
          <widget class="QPushButton" name="gen_MR_plot_button">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Generate MR-Error Recognition Ratio Graph</string>
           </property>
          </widget>
          <widget class="QPushButton" name="gen_arg_plot_button">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Generate MR Parameter-Error Recognition Ratio Graph</string>
           </property>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="checkbox_group">
   <property name="exclusive">
    <bool>false</bool>
   </property>
  </buttongroup>
 </buttongroups>
</ui>
